DirectProgramming/C++SYCL_FPGA/Tutorials/Features/multi_aocx/device_code_split/src/simple.cpp

```diff
@@ -0,0 +1,84 @@
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <iostream>
+
+using namespace sycl;
+constexpr access::mode sycl_read = access::mode::read;
+constexpr access::mode sycl_write = access::mode::write;
+
+class SimpleAdd;
+class SimpleMul;
+
+void kernelAdd(const float a, const float b, float &c) {
+
+  {
+#if FPGA_SIMULATOR
+    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+    auto selector = sycl::ext::intel::fpga_selector_v;
+#else // #if FPGA_EMULATOR
+    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+    queue q(selector);
+    buffer<float, 1> bufferA(&a, 1);
+    buffer<float, 1> bufferB(&b, 1);
+    buffer<float, 1> bufferC(&c, 1);
+
+    q.submit([&](handler &h) {
+      auto const accessorA = bufferA.template get_access<sycl_read>(h);
+      auto accessorB = bufferB.template get_access<sycl_read>(h);
+      auto accessorC = bufferC.template get_access<sycl_write>(h);
+      h.single_task<SimpleAdd>(
+          [=]() { accessorC[0] = accessorA[0] + accessorB[0]; });
+    });
+
+    q.wait();
+  }
+}
+
+void kernelMul(const float a, const float b, float &c) {
+
+  {
+#if FPGA_SIMULATOR
```

Comment:
Why doing the device selection in multiple places in the code?
Here it's clear that the same selector will be used for all kernels.
It is to show that different selectors can be used per kernel? 
If so a comment mentioning this would help (or remove the repetition of the selectors and just have a comment)